src/adapters/websocket-adapter.ts

```diff
@@ -0,0 +1,67 @@
+
+import { baseURL } from "../util/fetch";
+import {io , Socket} from 'socket.io-client'
+
+//New message adapter
+
+//By using interFace we are defining a structure for our object, here EventListener is an object with single optional message property, which return nothing.
+interface EventListeners {
+  newMessage?: (message: any) => void;
+  partnerUsers?: (activeUsers: any) => void;
+}
+/** 
+*@param {EventListeners} eventListeners - The server URL to be connected with.
+*@example
+*const url = `${baseURL}`
+*const eventListeners{
+*   newMessage : message => 
+*   console.log(`New message recieved ${message}`)
+*}
+*newMessageAdapter(eventListeners)
+**/
+export const newMessageAdapter = async(eventListeners: EventListeners) => {
+    
+    //creates a new instance of the Socket.IO client that is connected to the server
+    const socket = io(baseURL);
```

Comment:
Instead of baseURL doesn't the url need to be ws://localhost:8000 or ws://render.com?